src/components/navigation/section-document-or-browser.tsx

```diff
@@ -172,9 +182,102 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = observer(function Sect
       tabsExtraClassNames={{"chat-open": isChatOpen}}
       onSelect={handleTabSelect}
       selectedIndex={subTabIndex}
-      renderSubTabPanel={subTab =>
-        renderDocumentView(subTab) || renderDocumentBrowserView(subTab)
-      }
+      renderSubTabPanel={subTab => renderDocumentView(subTab) || renderDocumentBrowserView(subTab)}
     />
   );
 });
+
+interface DocumentBrowserScrollerProps {
+  subTab: ISubTabSpec;
+  tabSpec: NavTabModelType;
+  openDocumentKey: string;
+  onSelectDocument: (document: DocumentModelType) => void;
+}
+const DocumentBrowserScroller =
+    ({subTab, tabSpec, openDocumentKey, onSelectDocument}: DocumentBrowserScrollerProps) => {
+  const [scrollerCollapsed, setScrollerCollapsed] = useState(false);
+  const [collectionElement, setCollectionElement] = useState<HTMLDivElement>();
+  const documentScrollerRef = useRef<HTMLDivElement>(null);
+  const [scrollToLocation, setScrollToLocation] = useState(0);
+  const [panelWidth, setPanelWidth] = useState(0);
+
+  const scrollWidth = collectionElement?.scrollWidth ?? 0;
```

Comment:
```suggestion
  const scrollWidth = collectionElement?.scrollWidth ?? 0;
  const maxScrollTo = scrollWidth - panelWidth;
```
Then use `maxScrollTo` on lines 228 and 253.